File: File Explorer/LinkedList.java
class FileNode{
    String filename;
    int Filesize;
    FileNode next;

    public FileNode(String filename,int Filesize){
        this.filename=filename;
        this.Filesize=Filesize;
        this.next=null;
    }
}
class LinkedList{
    private FileNode head;

    public void createFile(String Filename,int FileSize){
        FileNode newnode=new FileNode(Filename,FileSize);
        if(head==null){
            head=newnode;
            return;
        }
        FileNode temp=head;
        while(temp.next!=null){
            temp=temp.next;
        }
        temp.next=newnode;
    }
    public void deleteFile(String filename){
        FileNode temp=head;
        if(head==null){
            System.out.println("There are no files in the directory");
            return;
        }
        if(head.filename.equals(filename)){
            head=head.next;
            System.out.println("The file is deleted succesfully");
            return;
        }
        while(temp.next!=null){
            if(temp.next.filename.equals(filename)){
                temp.next=temp.next.next;
                System.out.println("The file is deleted succesfully");
                return;
            }
            temp=temp.next;
        }
        System.out.println("The file is not found");
    }
    public void update_file(String old_filename,String new_filename){
        FileNode temp=head;
        while(temp != null){
            if(temp.filename.equals(old_filename)){
                temp.filename=new_filename;
                System.out.println("File name is updated");
                return;
            }
            temp=temp.next;
        }
        System.out.println("File is not found");
    }
    public void displayFiles(){
        if(head==null){
            System.out.println("There are no files");
            return;
        }
        FileNode temp=head;
        while(temp!=null){
            System.out.println("FileName: "+temp.filename+" File Size: "+temp.Filesize);
            temp=temp.next;
        }

    }
}

File: File Explorer/LinkedList.py
class FileNode:
    def __init__(self,filename,filesize):
        self.filename=filename
        self.filesize=filesize
        self.next=None
class FileManager:
    def __init__(self):
        self.head=None
    def create_file(self,filename,filesize):
        new_node=FileNode(filename,filesize)
        if self.head is None:
            self.head=new_node
            return
        temp_node=self.head
        while(temp_node.next):
            temp=temp.next
        temp.next=new_node
    def delete_file(self,filename):
        if self.head is None:
            print("No files are there to delete")
            return
        if self.head.filename==filename:
            self.head=self.head.next
            print("file deleted succesfully")
            return
        temp=self.head
        while(temp.next):
            if temp.next.filename==filename:
                temp=temp.next.next
                print("File deleted succesfully")
                return
            temp=temp.next
        print("File not found")
    def update_file(self,oldfilename,newfilename):
        temp=self.head
        while(temp):
            if(temp.filename==oldfilename):
                temp.filename=newfilename
                print("File name is updated succesfully")
                return 
            temp=temp.next
        print("file is not found")
    def display_files(self):
        if(self.head is None):
            print("No files are there")
        temp=self.head
        while(temp):
            print("File name:",temp.filename," File size:",temp.filesize)
            temp=temp.next
        
